catch-js-errors.js

```diff
@@ -0,0 +1,71 @@
+// Props http://stackoverflow.com/a/20405830/379063
+function stringifyErrorForUrlParams( error ) {
+	var simpleObject = {};
+	Object.getOwnPropertyNames( error ).forEach( function( key ) {
+		simpleObject[ key ] = encodeURIComponent( error[ key ] );
+	} );
+	return JSON.stringify( simpleObject );
+};
+
+// Props http://stackoverflow.com/a/17604754/379063
+function isLocalStorageNameSupported() {
+	var testKey = 'test',
+		storage = window.localStorage;
+	try {
+		storage.setItem( testKey, '1' );
+		storage.removeItem( testKey );
+		return true;
+	} catch ( error ) {
+		return false;
+	}
+}
+
+/**
+ * Throttle function by delay ms
+ *
+ * @param {function} fn wrapped function
+ * @param {number} delay minimum debounce time in ms
+ * @returns {function} a callback
+ */
+function debounce( fn, delay ) {
```

Comment:
Ideally we wouldn't leak generic names like this to the global scope. Either fitting into the current process or [wrapping it in an IIFE](http://benalman.com/news/2010/11/immediately-invoked-function-expression/) should work.